lib/publiccloud/provider.pm

```diff
@@ -255,13 +264,31 @@ sub create_instances {
     my ($self, %args) = @_;
     $args{check_connectivity} //= 1;
 
-    my @vms = $self->terraform_apply(%args);
-    foreach my $instance (@vms) {
-        record_info("INSTANCE", $instance->{instance_id});
-        if ($args{check_connectivity}) {
-            $instance->wait_for_ssh();
-            # Install server's ssh publicckeys to prevent authenticity interactions
-            assert_script_run(sprintf('ssh-keyscan %s >> ~/.ssh/known_hosts', $instance->public_ip));
+    my @vms;
+    if (get_var('PUBLIC_CLOUD_EXISTING_ADDRESS')) {
```

Comment:
Same as with `create_ssh_key`, overwrite the `create_instance()` method in `noprovider` class and do the special things there!